app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt

```diff
@@ -43,18 +64,84 @@ class CurrentFragment : Fragment() {
         }
 
         swipeRefreshLayout.setOnRefreshListener {
-            // TODO implement refresh usecase
+            presenter?.refresh(true) // TODO check for network state
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        presenter?.viewOnStop()
+    }
+
+    override fun onNavigationItemSelected(item: MenuItem): Boolean = when (item.itemId) {
+        R.id.itemCurrent -> {
+            drawerLayout.closeDrawers()
+            true
+        }
+        R.id.itemForecast -> {
+            // TODO navigate to ForecastFragment
+            true
+        }
+        R.id.itemHistory -> {
+            // TODO navigate to HistoryFragment
+            true
+        }
+        R.id.itemLocations -> {
+            // TODO navigate to LocationsFragment
+            true
+        }
+        R.id.itemSettings -> {
+            // TODO navigate to SettingsFragment
+            true
+        }
+        else -> false
+    }
+
+    override fun onRefreshSuccess() {
+        presenter?.let {
+            collapsingToolbar.title = it.city.name
+            textDateTime.text = it.currentWeatherDetails.dateTime.toString() // TODO convert Long to String
 
-            refreshFinish()
+            with(it.currentWeatherDetails) {
+                textCurrentTemperature.text = unitSystem.formatTemperature(temperature, resources)
+                textFeelsLike.text = unitSystem.formatTemperature(feelsLike, resources)
+                textWeatherDescription.text = weathers.getOrNull(0)?.description
+                    ?: getString(R.string.title_holder_description)
+                textVisibility.text = unitSystem.formatDistance(visibility, resources)
+                textWindSpeed.text = unitSystem.formatSpeed(windSpeed, resources)
+                textUVIndex.text = uvIndex?.toString() ?: getString(R.string.title_holder_number)
+                textPressure.text = unitSystem.formatPressure(pressure, resources)
+                textCloud.text = getString(R.string.format_percent_decimal, clouds)
+                textHumidity.text = getString(R.string.format_percent_decimal, humidity)
+            }
+            with(it.dailyWeather) {
+                textSunrise.text = sunrise.toString()
+                textDayTemperature.text = unitSystem.formatTemperature(temperature.day, resources)
+                textSunset.text = sunset.toString()
+                textNightTemperature.text =
+                    unitSystem.formatTemperature(temperature.night, resources)
+            }
+
+            (recyclerTodaySummaryWeather.adapter as? SummaryWeatherAdapter)?.submitList(it.todaySummaryWeather)
+            (recyclerForecastSummaryWeather.adapter as? SummaryWeatherAdapter)?.submitList(it.forecastSummaryWeather)
         }
+        refreshFinished()
+    }
+
+    override fun onRefreshError(exception: Exception) {
+        Toast.makeText(requireContext(), exception.message, Toast.LENGTH_SHORT).show()
```

Comment:
ô viết ToastUtils.kt cho trường hợp này để sử dụng nhiều nơi cho tiện nhé

Reply:
viết context extentsion cho toast cx được 
`fun Context.showToast(msg: String) {
    Toast.makeText(this, msg,Toast.LENGTH_LONG).show()
}`